metricflow/sql_clients/databricks.py

```diff
@@ -0,0 +1,177 @@
+        logger.info("Completed conversion of PyArrow Table to pandas DataFrame.")
+        return pandas_df
+
+    def _engine_specific_execute_implementation(self, stmt: str, bind_params: SqlBindParameters) -> None:
+        """Execute statement, returning nothing."""
+        with self.get_connection() as connection:
+            with connection.cursor() as cursor:
+                logger.info(f"Executing SQL statment: {stmt}")
+                cursor.execute(operation=stmt, parameters=self.params_or_none(bind_params))
+
+    def _engine_specific_dry_run_implementation(self, stmt: str, bind_params: SqlBindParameters) -> None:
+        """Check that query will run successfully without actually running the query, error if not."""
+        stmt = f"EXPLAIN {stmt}"
+
+        with self.get_connection() as connection:
+            with connection.cursor() as cursor:
+                logger.info(f"Executing SQL statment: {stmt}")
+                cursor.execute(operation=stmt, parameters=self.params_or_none(bind_params))
+
+                # If the plan contains errors, they won't be raised. Parse plan string to find & raise errors.
+                result = str(cursor.fetchall_arrow()["plan"][0])
+                if "org.apache.spark.sql.AnalysisException" in result:
+                    error = result.split("== Physical Plan ==")[1].split(";")[0]
+                    raise sql.exc.ServerOperationError(error)
+
+    def create_table_from_dataframe(  # noqa: D
+        self, sql_table: SqlTable, df: pd.DataFrame, chunk_size: Optional[int] = None
+    ) -> None:
+        logger.info(f"Creating table '{sql_table.sql}' from a DataFrame with {df.shape[0]} row(s)")
+        start_time = time.time()
+        with self.get_connection() as connection:
+            with connection.cursor() as cursor:
+                # Create table
+                columns = df.columns
+                columns_to_insert = []
+                for i in range(len(df.columns)):
+                    # Format as "column_name column_type"
+                    columns_to_insert.append(f"{columns[i]} {PANDAS_TO_SQL_DTYPES[str(df[columns[i]].dtype)]}")
+                cursor.execute(f"CREATE TABLE IF NOT EXISTS {sql_table.sql} ({', '.join(columns_to_insert)})")
+
+                # Insert rows
+                values = []
+                for row in df.itertuples(index=False, name=None):
+                    cells = []
+                    for cell in row:
+                        if type(cell) in [str, pd.Timestamp]:
+                            # Wrap cell in quotes & escape existing single quotes
+                            escaped_cell = str(cell).replace("'", "\\'")
+                            cells.append(f"'{escaped_cell}'")
+                        else:
+                            cells.append(str(cell))
+                    values.append(f"({', '.join(cells)})")
+                cursor.execute(f"INSERT INTO {sql_table.sql} VALUES {', '.join(values)}")
```

Comment:
I see what you mean about this being a bit hacky. I don't understand why they don't support creating tables from arrow or just dataframe -> sql. Might be worth opening an issue over there.